torch/utils/tensorboard/writer.py

```diff
@@ -3,9 +3,11 @@
 
 import os
 import time
+from typing import List, Optional, Union
 
 import torch
 
+from matplotlib.figure import Figure
```

Comment:
I don't think this is a hard dependency in Pytorch is it? If not we probably need to hide it behind a TYPE_CHECKING if.